src/driver/socket.rs

```diff
@@ -31,6 +32,14 @@ impl Socket {
         Ok(Socket { fd })
     }
 
+    pub(crate) fn new_unix(socket_type: libc::c_int) -> io::Result<Socket> {
+        let socket_type = socket_type | libc::SOCK_CLOEXEC;
+        let domain = libc::AF_UNIX;
+        let fd = syscall!(socket(domain, socket_type, 0))?;
```

Comment:
You can probably do this with socket2

Reply:
@jon-chuang can we do this without the raw syscall here?

Reply:
My bad, missed this one